pypsa_to_IPCC.py

```diff
@@ -121,136 +121,226 @@
+                 *n.links.p_nom_opt.filter(like ='solid biomass CHP CC').filter(like =country).filter(like=str(dict_var[v_type]))).sum())   
+               var[v_type+'|Electricity|Biomass|w/o CCS'] = (var[v_type+'|Electricity|Biomass']    
+              -var[v_type+'|Electricity|Biomass|w/ CCS'])
+            
+               var[v_type+'|Electricity|Oil'] = 0.001*((n.links.efficiency.filter(like ='oil-').filter(like =country).filter(like=str(dict_var[v_type]))
+                 *n.links.p_nom_opt.filter(like ='oil-').filter(like =country).filter(like=str(dict_var[v_type]))).sum())
+            
+               """
+               Capacity : hydro (reservoir, ror)
+               """
+               #MW -> GW
+               var[v_type+'|Electricity|Hydro'] = 0.001*n.generators.p_nom_opt.filter(like ='ror').filter(like =country).filter(like=str(dict_var[v_type])).sum()
+               var[v_type+'|Electricity|Hydro'] += 0.001*n.storage_units.p_nom_opt.filter(like ='hydro').filter(like =country).filter(like=str(dict_var[v_type])).sum()
+               """
+               Capacity : storage (PHS, battery, H2 storage, thermal storage)
+               """
+               #MWh to GWh
+               var[v_type+'|Electricity|Storage|Pumped Hydro Storage'] = 0.001*(n.storage_units.p_nom_opt.filter(like ='PHS').filter(like =country).filter(like=str(dict_var[v_type])).sum())
+               var[v_type+'|Electricity|Storage|Battery Capacity|Home Battery'] = 0.001*((n.links.efficiency.filter(like ='home battery charger').filter(like =country).filter(like=str(dict_var[v_type]))
+                 *n.links.p_nom_opt.filter(like ='home battery charger').filter(like =country).filter(like=str(dict_var[v_type]))).sum())
+               var[v_type+'|Electricity|Storage|Battery Capacity'] = 0.001*((n.links.efficiency.filter(like ='battery charger').filter(like =country).filter(like=str(dict_var[v_type]))
+                 *n.links.p_nom_opt.filter(like ='battery charger').filter(like =country).filter(like=str(dict_var[v_type]))).sum())
+               var[v_type+'|Electricity|Storage|Battery Capacity|Utility-scale Battery'] = (var[v_type+'|Electricity|Storage|Battery Capacity'] 
+                 - var[v_type+'|Electricity|Storage|Battery Capacity|Home Battery'])
+               #var[v_type+'|Electricity|Storage|Hydrogen Storage Capacity|overground'] = 0.001 *(n.stores.e_nom_opt.filter(like ='H2').filter(like =country).filter(like=str(dict_var[v_type])).sum()/168) #assume one week charge time for H2 storage
+               #var[v_type+'|Electricity|Storage|Hydrogen Storage Capacity|underground'] = 0.001 *n.stores.e_nom_opt[country + ' H2 Store underground'] if country + ' H2 Store underground' in n.stores.index else 0
+               var[v_type+'|Electricity|Storage|Hydrogen Storage Capacity'] = 0.001 *(n.stores.e_nom_opt.filter(like ='H2').filter(like =country).filter(like=str(dict_var[v_type])).sum()/168)
```

Comment:
this reports the power capacity, to report the energy capacity no need to divide by 168

also, let's avoid hard-coding numbers, this can be read from the attribute max_hours

Reply:
I removed the 168 for the energy capacity. As for power capacity (added in 'Capacity|Electricity|Storage Capacity'), I'm now using the fuel cell link capacity as the power capacity of hydrogen storage.

Reply:
This is now deleted since it does not match the rest of the storage capacities with unit GWh